### PR TITLE
Code cleanup : modernize switch expressions

### PR DESCRIPTION
**Pull request by elsazac**
Code cleanup : modernize switch expressions

### CODE REVIEW

File: bundles/org.eclipse.swt/Eclipse SWT Accessibility/cocoa/org/eclipse/swt/accessibility/Accessible.java [hunk 1]
```diff
@@ -1671,20 +1671,15 @@ id getAttributedStringForRangeParameterizedAttribute(id parameter, int childID)
...
 					NSNumber styleObj = null;
-					switch (style) {
-					case SWT.UNDERLINE_SINGLE:
-						styleObj = NSNumber.numberWithInt(OS.kAXUnderlineStyleSingle);
-						break;
-					case SWT.UNDERLINE_DOUBLE:
-						styleObj = NSNumber.numberWithInt(OS.kAXUnderlineStyleDouble);
-						break;
-					case SWT.UNDERLINE_SQUIGGLE:
+					styleObj = switch (style) {
```
Comment (mickaelistria) [inline bundles/org.eclipse.swt/Eclipse SWT Accessibility/cocoa/org/eclipse/swt/accessibility/Accessible.java:1674]: Could those 2 lines be joined?

File: bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/widgets/Tracker.java [hunk 2]
```diff
@@ -250,35 +250,17 @@ Point adjustResizeCursor (boolean movePointer) {
...
+		case SWT.UP -> new Cursor(display, SWT.CURSOR_SIZENS);
+		case SWT.DOWN -> new Cursor(display, SWT.CURSOR_SIZENS);
+		case SWT.LEFT -> new Cursor(display, SWT.CURSOR_SIZEWE);
+		case SWT.RIGHT -> new Cursor(display, SWT.CURSOR_SIZEWE);
+		case SWT.LEFT | SWT.UP -> new Cursor(display, SWT.CURSOR_SIZENWSE);
+		case SWT.RIGHT | SWT.DOWN -> new Cursor(display, SWT.CURSOR_SIZENWSE);
+		case SWT.LEFT | SWT.DOWN -> new Cursor(display, SWT.CURSOR_SIZENESW);
+		case SWT.RIGHT | SWT.UP -> new Cursor(display, SWT.CURSOR_SIZENESW);
+		default -> new Cursor(display, SWT.CURSOR_SIZEALL);
+		};
```
Comment (mickaelistria) [inline bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/widgets/Tracker.java:263]: Could be something like
```java
Cursor newCursor = new Cursor(display, switch (cursorOrientation) {
  case SWT.UP, SWT.DOWN -> SWT.CURSOR_SIZENS;
  case SWT.LEFT, SWT.RIGHT -> SWT.CURSOR_SIZEWE;
  case (SWT.LEFT | SWT.UP), (SWT.RIGHT | SWT.DOWN ) -> new Cursor(display, SWT.CURSOR_SIZENWSE);
...
});
```

File: bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/DPIUtil.java [hunk 3]
```diff
@@ -251,11 +249,11 @@ private static ImageData autoScaleImageData (Device device, final ImageData imag
...
 		ImageData result = resultImage.getImageData (DPIUtil.getDeviceZoom ());
 		resultImage.dispose ();
-		return result;
-	case NEAREST:
-	default:
-		return imageData.scaledTo (scaledWidth, scaledHeight);
+		yield result;
 	}
+	case NEAREST -> imageData.scaledTo (scaledWidth, scaledHeight);
+	default -> imageData.scaledTo (scaledWidth, scaledHeight);
```
Comment (mickaelistria) [inline bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/DPIUtil.java:255]: `default` should be sufficient t cover both cases

File: bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/JPEGDecoder.java [hunk 4]
```diff
@@ -2462,22 +2462,18 @@ static void jpeg_calc_output_dimensions (jpeg_decompress_struct cinfo)
...
-		case JCS_YCCK:
-			cinfo.out_color_components = 4;
-			break;
-		default:			/* else must be same colorspace as in file */
-			cinfo.out_color_components = cinfo.num_components;
-			break;
-	}
+	cinfo.out_color_components = switch (cinfo.out_color_space) {
+	case JCS_GRAYSCALE:
+		yield 1;
```
Comment (mickaelistria) [inline bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/JPEGDecoder.java:2467]: Why not `case JCS_GRAYSCALE -> 1;` ?

File: bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/PngEncoder.java [hunk 5]
```diff
@@ -237,20 +237,12 @@ void writeImageData() throws IOException {
...
-		os = new DeflaterOutputStream(baos, new Deflater(BEST_SPEED));
-		break;
-	case 3:
-		os = new DeflaterOutputStream(baos, new Deflater(BEST_COMPRESSION));
-		break;
-	default:
-		os = new DeflaterOutputStream(baos, new Deflater(DEFAULT_COMPRESSION));
-		break;
-	}
+	os = switch (loader.compression) {
```
Comment (mickaelistria) [inline bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/PngEncoder.java:240]: Those lines could be joined,

Comment (mickaelistria) [reply bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/PngEncoder.java:240]: Or even better
```java
OutputStream os = new DeflaterOutputStream(baos, switch (loader.compression) {
case 0 -> NO_COMPRESSION;
case 1 -> BEST_SPEED;
case 3 -> BEST_COMPRESSION;
default -> DEFAULT_COMPRESSION;
}
```

File: bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/JPEGDecoder.java (hunk 4 above)
Comment (elsazac) [reply bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/JPEGDecoder.java]: when I applied your changes I was getting this error "mixing of different case semantics is not allowed". I converted all the cases into one single format. Not sure if I did it right. Can you suggest?

File: bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/widgets/Tracker.java (hunk 2 above)
Comment (elsazac) [reply bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/widgets/Tracker.java]: addressed the changes.

File: bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/DPIUtil.java (hunk 3 above)
Comment (elsazac) [reply bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/DPIUtil.java]: addressed the changes

File: bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/PngEncoder.java (hunk 5 above)
Comment (elsazac) [reply bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/PngEncoder.java]: adressed the changes.

File: bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/JPEGDecoder.java (hunk 4 above)
Comment (mickaelistria) [reply bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/JPEGDecoder.java]: the `->` syntax must be used on all fields; it's not allowed to mix `case X: ...` and `case Y -> ...`

Comment (elsazac) [reply bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/internal/image/JPEGDecoder.java]: I have addressed that way.

**Comment by mickaelistria** [comment]
Thank you!

**Pull request merged**